### PR TITLE
[Merged by Bors] - feat: port LazyList

### PR DESCRIPTION
**Pull request by siddhartha-gadgil**
from commit c3019c79074b0619edb4b27553a91b2e82242395

### CODE REVIEW

File: Mathlib/Data/LazyList.lean [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+-/
+import Mathlib.Mathport.Rename
+/-!
+# Lazy lists
+
+The type `lazy_list α` is a lazy list with elements of type `α`.
+In the VM, these are potentially infinite lists
+where all elements after the first are computed on-demand.
+(This is only useful for execution in the VM,
+logically we can prove that `lazy_list α` is isomorphic to `list α`.)
```
Comment (semorrison) [inline Mathlib/Data/LazyList.lean:14]:
```suggestion
The type `LazyList α` is a lazy list with elements of type `α`.
In the VM, these are potentially infinite lists
where all elements after the first are computed on-demand.
(This is only useful for execution in the VM,
logically we can prove that `LazyList α` is isomorphic to `List α`.)
```

File: Mathlib/Data/LazyList.lean [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  | cons a _, 0 => some a
+  | cons _ l, n + 1 => nth (l.get) n
+#align lazy_list.nth LazyList.nth
+
+/-- The infinite lazy list `[x, f x, f (f x), ...]` of iterates of a function.
+This definition is meta because it creates an infinite list.
+-/
+unsafe def iterates (f : α → α) : α → LazyList α
+  | x => cons x (iterates f (f x))
+-- #align lazy_list.iterates lazy_list.iterates
```
Comment (semorrison) [inline Mathlib/Data/LazyList.lean:139]: Why are these commented out?

Comment (siddhartha-gadgil) [reply Mathlib/Data/LazyList.lean:139]: I was getting an error `Declaration lazy_list.iterates not found` with the suggestion of adding `set_option align.precheck false`

Comment (semorrison) [reply Mathlib/Data/LazyList.lean:139]: Ah: it needs to be `#align lazy_list.iterates LazyList.iterates`: it is correctly telling you that you mis-spelled the second argument!

**Comment by semorrison** [comment]
bors d+

after fixing the #aligns. Please update the port-status page after merging.

File: Mathlib/Data/LazyList.lean (hunk 2 above)
Comment (siddhartha-gadgil) [reply Mathlib/Data/LazyList.lean:139]: Thanks. Will fix

**Pull request closed without merging**